### PR TITLE
WINDUP-997: spring-mvc.xml as invalid by 'XML File is not valid' rule

### PR DESCRIPTION
**Pull request by jsight**
https://issues.jboss.org/browse/WINDUP-997

### CODE REVIEW

File: rules-xml/impl/src/main/java/org/jboss/windup/rules/apps/xml/xml/ValidateXmlFilesRuleProvider.java [hunk 1]
```diff
@@ -141,35 +152,72 @@ private FileLocationModel createLocationModelFromNodeElement(Node node, GraphCon
...
-                return false;
+                parseFailed = true;
+            }
+
+            if (parseFailed)
+            {
+                sourceFile.setGenerateSourceReport(true);
+
+                ClassificationService classificationService = new ClassificationService(event.getGraphContext());
+                ClassificationModel model =classificationService.attachClassification(context, sourceFile, XmlFileModel.NOT_VALID_XML,
```
Comment (OndraZizka) [inline rules-xml/impl/src/main/java/org/jboss/windup/rules/apps/xml/xml/ValidateXmlFilesRuleProvider.java]: Missing space :)

Comment (OndraZizka) [inline rules-xml/impl/src/main/java/org/jboss/windup/rules/apps/xml/xml/ValidateXmlFilesRuleProvider.java]: Missing space :)

**Comment by OndraZizka** [comment]
We have the Unparsable files report, does this count with it? I know that if it doesn't, the original did not either, but once on it, we should add it to the unparsables.
The way to mark it so is to add set the `parseError` property.

**Comment by jsight** [comment]
I don't think we have generally treated a xsd validation error as "unparsable". In many cases these files are actually valid and working within their context, which isn't really the same as unparsable.

**Pull request merged**